packages/flutter/lib/src/widgets/app.dart

```diff
@@ -231,11 +290,25 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
   GlobalObjectKey<NavigatorState> _navigator;
   Locale _locale;
 
+  Locale _resolveLocale(Locale newLocale, Iterable<Locale> supportedLocales) {
+    if (widget.resolveLocaleCallback != null)
+      return widget.resolveLocaleCallback(newLocale, widget.supportedLocales);
+
+    Locale matchesCountryCode;
```

Comment:
this variable's name is quite confusing since it's actually the locale that didn't match the country code...